Sprint-1/implement/max.js

```diff
@@ -1,4 +1,10 @@
-function findMax(elements) {
-}
-
-module.exports = findMax;
+function findMax(arr) {
+    if (!Array.isArray(arr)) {
+      throw new TypeError("Input must be an array");
+    }
+    const numericValues = arr.filter(item => typeof item === 'number');
+    return numericValues.length > 0 ? Math.max(...numericValues) : -Infinity;
```

Comment:
What do you expect from the following function calls (on extreme cases)?
Does your function returns the value you expected?

```
findMax([-Infinity])
findMax([-Infinity, 0])
findMax([NaN])
findMax([0, NaN, 1])
```

Reply:
findMax([-Infinity])

Expected: -Infinity
Reasoning: Only -Infinity in the array, so it’s the maximum.
findMax([-Infinity, 0])

Expected: 0
Reasoning: The maximum between -Infinity and 0 is 0.
findMax([NaN])

Expected: NaN
Reasoning: NaN is not comparable, so it stays as the maximum.
findMax([0, NaN, 1])

Expected: 1
Reasoning: NaN is ignored, so the maximum is 1.